pkg/datasources/database_role.go

```diff
@@ -0,0 +1,74 @@
+package datasources
+
+import (
+	"context"
+	"log"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/internal/provider"
+
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/sdk"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+var databaseRoleSchema = map[string]*schema.Schema{
+	"database": {
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The database from which to return the database role from.",
+	},
+	"role": {
```

Comment:
Please change the `role` field to `name`, so it will match with other data sources.